### ISSUE TITLE
laceOS Package Suggestions

### ISSUE THREAD

**Issue by charlie5** [charlie5/laceOS#2]
Add Archlinux packages to include in laceOS here.

**Comment by hurufu** [comment]
Thanks for your effort of putting all useful Ada libraries together! Is it possible to release just a metapackage in AUR that will install all necessary prerequisites for Ada development? Without downloading a VM with whole OS inside to just develop couple programs in Ada?

**Comment by charlie5** [comment]
@hurufu
Yes, I've been considering an Ada meta-package for the AUR. I'll try to get it done this weekend. It will be called 'all-ada'.

'laceOS' isn't a VM, although it can be installed into one. It is an Archlinux installer with all the Ada tools included, along with supporting tools and a desktop configured in a manner conducive to s/w development. It's main target audience is people wishing to try/use Ada, who are not already using Archlinux.

I'm also looking into creating an Ada Arch Repository which can be added to pacman's repository list, which contains the Ada packages prepackaged. Again, I will try to get that going this weekend, also.


Regards.

**Comment by hurufu** [comment]
Thanks a lot for creating that metapackage, I've installed `all-ada` package from your repo and everything looks good.
Sometimes gnatprove is unable to prove statements about my code and crashes, but I think it is a problem to be reported to GNAT developers.

Also do you know how to build standard Ada runtime to have better description of what was the cause of an exception?  Message `raised CONSTRAINED_ERROR`  isn't really helping when you try to debug your program.

**Comment by charlie5** [comment]
@hurufu

You're welcome :).

Try adding '-gnateE' to your compiler switches to add extra debug info in 'constraint_Error' messages.

You should then get messages like ...

Constraint_Error value 0 not in range 1 .. 5

I and many others are in the #ada channel on libera irc server. You can get (almost) instant help there for many Ada issues. 


Regards.
